backend/frontend/urls.py

```diff
@@ -68,7 +68,7 @@
     ),
     # Course component details
     path(
-        "project/<int:project_id>",
+        "course_component_details/<int:id>/<str:type>",
```

Comment:
The URLs can be duplicated. So have the `/project/id` and `/topic/id` back, and refer them to the same view. Then in the view, you can check if the request URL is from a topic or project and fetch the necessary id.

I think it should be possible.

Duplication might not be a bad idea here. I'm just not confident which places it should and shouldn't be the way to go at the moment. But this seems easier and more approachable than adding another `<str:type>`